docs/docs/code/bash/README.md

```diff
@@ -0,0 +1,199 @@
+
+This enables you to only worry about the body of the response so you can visualize it with tools like `echo` or `jq`. 
+:::
+
+### Making a POST request
+
+`curl` handles making POSTs requests as well. The `-X` flag allow you to specify the HTTP method you'd like to use for an HTTP request.
+
+The `-d` flag is for passing data in the POST request.
+
+```bash
+curl --silent -X POST https://postman-echo.com/post -d 'name=Bulbasaur&id=1'
+
+# to store the API response in a variable, interpolate the response into a string and store it in variable
+RESPONSE=`curl --silent -X POST https://postman-echo.com/post -d 'name=Bulbasaur&id=1'`
+
+# now the response is stored as a variable
+echo $RESPONSE
+```
+
+### Using API key authentication
+
+Some APIs require you to use a secret API key to prove your access.
+
+`curl` has an `-h` flag where you can pass your API key as a token.
+
+Below is an example of searching Twitter for mentions of a specific Twitter handle. Twitter requires a valid API key passed in the header of the request.
+
+```bash
+curl --silent -X POST -h "authorization:Bearer $(<your api key here>)" https://api.twitter.com/2/users/@pipedream/mentions
+```
+
+## Sharing data between steps
+
+A step can accept data from other steps in the same workflow, or pass data downstream to others.
+
+This makes your steps even more powerful, you can compose new workflows and reuse steps.
+
+### Using data from another step
+
+In bash scripts, data from the initial workflow trigger and other steps are available in the `$PIPEDREAM_STEPS` environment variable.
+
+In this example, we'll pretend this data is coming into our HTTP trigger via a POST request.
+
+```json
+{
+  "id": 1,
+  "name": 'Bulbasaur",
+  "type": "plant"
+}
+```
+
+In our bash script, we can access this data via the `$PIPEDREAM_STEPS` variable. Specifically, this data from the POST request into our workflow is available in the `trigger` object.
+
+```bash
+echo $PIPEDREAM_STEPS | jq .trigger.event
+
+# results in { id: 1, name: "Bulbasaur", type: "plant" }
+```
+
+::: tip
+The period (`.`) in front the `trigger.event` in the example is not a typo. This is to define the starting point for `jq` to start traversing down the JSON in the HTTP response.
+:::
+
+### Sending data downstream to other steps
+
+To share data created, retrieved, transformed or manipulated by a step to others downstream, append it to the `$PIPEDREAM_EXPORTS` variable.
+
+An example speaks a thousand words, so here's one passing data from an API to the bash step.
+
+```bash
+# Retrieve the data from an API and store it in a variable
+DATA=`curl --silent https://pokeapi.co/api/v2/pokemon/charizard`
+
+# Write data to $PIPEDREAM_EXPORTS to share with steps downstream
+EXPORT="key:json=${DATA}"
+echo $EXPORT >> $PIPEDREAM_EXPORTS
+```
+
+::: tip
+Don't worry, the special `key` string in the `EXPORT` will automatically reference the current step's name. 
```

Comment:
This is actually not true. `key` refers to the actual name of the property exported from the step. If I set it to `key`:

<img width="242" alt="Screen Shot 2022-01-21 at 2 42 08 PM" src="https://user-images.githubusercontent.com/242668/150609713-22b86521-5190-4599-a74c-f9412bc20e34.png">

If I set to `foo:json=...`:

<img width="191" alt="Screen Shot 2022-01-21 at 2 40 23 PM" src="https://user-images.githubusercontent.com/242668/150609741-ba7120f5-fe9e-46f3-a9e7-2e89803d83e0.png">

In the context of the step, we always know `$PIPEDREAM_EXPORTS` refers to exports for the current step (e.g. `steps.code`). The key you pass at the beginning just defines the property under that (e.g. `steps.code.key`).

It's also worth noting that the `:json` after the key actually tells the execution environment that the data is JSON and that we should format it as a structured object in exports.

If I just do `key=value`, this is what the export looks like:

<img width="630" alt="Screen Shot 2022-01-21 at 2 38 22 PM" src="https://user-images.githubusercontent.com/242668/150609897-a59841ba-f465-4200-848d-7ca890d8710b.png">

When I do `key:json=value`, I get:

<img width="521" alt="Screen Shot 2022-01-21 at 2 38 05 PM" src="https://user-images.githubusercontent.com/242668/150609943-47b7ac8b-f275-4139-aa47-f432a3a98320.png">